Mapify/Locale.cs

```diff
@@ -2,70 +2,129 @@
             return true;
         }
 
+        public static void LoadMapCSV(string mapDir)
+        {
+            var localeFilePath = Path.Combine(mapDir, LOCALE_FILE);
+            if (!File.Exists(localeFilePath))
+            {
+                Mapify.LogDebug(nameof(LoadMapCSV)+$" no map CSV at {localeFilePath}");
+                return;
+            }
+
+            mapSpecificTranslations = CSV.Parse(File.ReadAllText(localeFilePath));
+            mapSpecificTranslationsLoaded = true;
+
+            Mapify.LogDebug(nameof(LoadMapCSV)+$" loaded map CSV at {localeFilePath}");
+        }
+
+        public static void UnloadMapCSV()
+        {
+            mapSpecificTranslations = new (new Dictionary<string, Dictionary<string, string>>());
+            mapSpecificTranslationsLoaded = false;
+        }
+
         public static string Get(string key)
         {
             if (!initializeAttempted)
                 throw new InvalidOperationException("Not initialized");
 
+            if (TryGetMapSpecificTranslation(key, out var translation))
+            {
+                return translation;
+            }
+
             string locale = LocalizationManager.CurrentLanguage;
 
-            if (!csv.ContainsKey(locale))
+            if (!mapifyTranslations.ContainsKey(locale))
             {
                 if (locale == DEFAULT_LANGUAGE)
                 {
                     Mapify.LogError($"Failed to find locale language {locale}! Something is broken, this shouldn't happen. Dumping CSV data:");
-                    Mapify.LogError($"\n{CSV.Dump(csv)}");
+                    Mapify.LogError($"\n{CSV.Dump(mapifyTranslations)}");
                     return MISSING_TRANSLATION;
                 }
 
+                Mapify.LogWarning($"Failed to find locale language '{locale}', using default language '{DEFAULT_LANGUAGE}'");
                 locale = DEFAULT_LANGUAGE;
-                Mapify.LogWarning($"Failed to find locale language {locale}");
             }
 
-            Dictionary<string, string> localeDict = csv[locale];
+            Dictionary<string, string> localeDict = mapifyTranslations[locale];
 
             if (localeDict.TryGetValue(key.TrimStart(PREFIX_CHARS), out string value)) {
                 return value;
             }
 
             // If there is no translation for this station's name, don't translate it.
-            if (key.StartsWith(STATION_PREFIX)) {
-                return key.Replace(STATION_PREFIX, "");;
+            if (key.StartsWith(STATION_PREFIX))
+            {
+                var stationID = key.Replace(STATION_PREFIX, "");
+                var stationName = StationController.allStations
+                    .Select(stationController => stationController.stationInfo)
+                    .Where(stationInfo => stationInfo.YardID == stationID)
+                    .Select(stationInfo => stationInfo.Name)
+                    .ToList()
+                    .FirstOrDefault();
```

Comment:
Does this get called on any hot paths?

Reply:
No, just when you open the handheld map or when a job booklet spawns